### DISCUSSION TITLE
esp32 use emodbus to control  BLDC motor driver

### DISCUSSION THREAD

**Discussion by zouzhe1** [eModbus/eModbus#305] [Q&A]
my motor driver is support rs485 modbus control
so  i  use    platformio   esp32   emodbus  write some code
It can work,but when i want to send and receive  modbus ,Some time later.
The program starts reporting errors：
**case REQUEST_QUEUE_FULL    : // 0xE8,
      return "Request queue full";**
I send one every 1ms（slave-id, WRITE_HOLD_REGISTER, uint32 address, uint32 speed）
**How can I solve this problem?** 
I increased the queueLimit from 100 to 1000 and there will still be problems. 
My baud rate is 115200

**Comment by Miq1** [comment]
1ms? That is far beyond the capabilities of the ESP32 I am afraid. You are sending 12 bytes at least (message plus CRC), that will use 120 bits, which again at 115200 baud will take more than 1ms to be transmitted, leave alone there has been no processing or a response in the calculation.

By the way: ``uint32`` for address and speed is looking suspicious as well, as normally a Modbus register is addressed with ``uint16_t``, as is a register's size as well.
